pathio.go

```diff
@@ -140,3 +138,23 @@ func getRegionForBucket(name string) (string, error) {
 	}
 	return *resp.LocationConstraint, nil
 }
+
+type s3Api interface {
+	GetBucketLocation(input *s3.GetBucketLocationInput) (*s3.GetBucketLocationOutput, error)
+}
+
+type liveS3Api struct {
+	liveS3 *s3.S3
+}
+
+func (m *liveS3Api) GetBucketLocation(input *s3.GetBucketLocationInput) (*s3.GetBucketLocationOutput, error) {
+	return m.liveS3.GetBucketLocation(input)
+}
+
+func getS3Api() *liveS3Api {
```

Comment:
In line with `go` conventions this should probably be called `newS3API`, or actually `newS3Handler`